### PR TITLE
Retrieve inspection specification from a Workflow

### PR DESCRIPTION
**Pull request by CermakM**
Depends-On: #265

### CODE REVIEW

File: amun/api_v1.py [hunk 1]
```diff
@@ -108,17 +119,23 @@ def _adjust_default_requests(dict_: dict) -> None:
     dict_['requests'] = new_requests
 
 
+@versionchanged(
+    version="0.6.0",
+    reason="The function now submits an Argo Workflow."
+)
```
Comment (fridex) [inline amun/api_v1.py]: Why is this needed?

File: amun/api_v1.py [hunk 2]
```diff
@@ -142,39 +159,69 @@ def post_inspection(specification: dict) -> tuple:
...
-            parameters,
-            use_hw_template=use_hw_template,
-            memory_requests=run_memory_requests,
-            cpu_requests=run_cpu_requests
-        )
+    template_parameters = dict(parameters)
+    template_parameters['AMUN_CPU'] = specification['build'].get(
+        'requests', {}).get('cpu')
+    template_parameters['AMUN_MEMORY'] = specification['build'].get(
+        'requests', {}).get('memory')
```
Comment (fridex) [inline amun/api_v1.py]: Please rebase, this has changed in the master.

File: amun/api_v1.py [hunk 3]
```diff
@@ -142,39 +159,69 @@ def post_inspection(specification: dict) -> tuple:
...
+    template_parameters['THOTH_INFRA_NAMESPACE'] = _OPENSHIFT.amun_infra_namespace
+
+    target = "inspection-run-result" if run_job else "inspection-build-result"
+
+    formated_dockerfile = dockerfile.replace("'", "''")
```
Comment (fridex) [inline amun/api_v1.py]: formatted

File: amun/api_v1.py [hunk 4]
```diff
@@ -273,22 +331,33 @@ def get_inspection_status(inspection_id: str) -> tuple:
...
             'parameters': parameters
         }
 
-    specification = json.loads(build['metadata']['annotations']['amun_specification'])
+    parameters: List[Dict[str, Any]] = wf["spec"]["arguments"]["parameters"]
+
+    specification, = map(
+        lambda s: s["value"],
+        filter(lambda p: p["name"] == "specification", parameters)
+    )
```
Comment (fridex) [inline amun/api_v1.py]: What is this used for?

File: amun/api_v1.py (hunk 1 above)
Comment (CermakM) [reply amun/api_v1.py]: Ah, right. I can remove these already for the updated functions.

File: amun/api_v1.py (hunk 2 above)
Comment (CermakM) [reply amun/api_v1.py]: This is a subject of the #265 PR, I will rebase it there.

File: amun/api_v1.py (hunk 3 above)
Comment (CermakM) [reply amun/api_v1.py]: This is a subject of the #265 PR, I will fix it there.

File: amun/api_v1.py (hunk 4 above)
Comment (CermakM) [reply amun/api_v1.py]: Can you elaborate a bit please?

Comment (fridex) [reply amun/api_v1.py]: The code is hard to read for me - too much map-lambda-filter.

**Comment by CermakM** [comment]
Closing, merged into #265
